crates/holochain/src/conductor/conductor.rs

```diff
@@ -849,6 +850,74 @@ mod network_impls {
             .collect::<Result<Vec<_>, _>>()
         }
 
+        pub(crate) async fn storage_info_for_app(
+            &self,
+            installed_app_id: &InstalledAppId,
+        ) -> ConductorResult<AppStorageInfo> {
+            let state = self.get_state().await?;
+            let app = state.get_app(installed_app_id)?;
+
+            let db_error_mapper = |e| ConductorError::DatabaseError(e);
+
+            let cell_storage_info =
+                futures::future::join_all(app.all_cells().map(|cell_id| async move {
+                    let authored_db = self.spaces.authored_db(cell_id.dna_hash())?;
+                    let dht_db = self.spaces.dht_db(cell_id.dna_hash())?;
+                    let cache_db = self.spaces.cache(cell_id.dna_hash())?;
```

Comment:
these DBs are all per-DNA and can return duplicate info

Reply:
This has now been reworked to put blobs (dna) at the top level as discussed on the linked issue